library/c/api_relay.go

```diff
@@ -4,58 +4,74 @@ package main
 #include <cgo_utils.h>
 */
 import "C"
-import "github.com/waku-org/go-waku/library"
+import (
+	"unsafe"
+
+	"github.com/waku-org/go-waku/library"
+)
 
 // Determine if there are enough peers to publish a message on a topic. Use NULL
 // to verify the number of peers in the default pubsub topic
 //
 //export waku_relay_enough_peers
-func waku_relay_enough_peers(topic *C.char, onOkCb C.WakuCallBack, onErrCb C.WakuCallBack) C.int {
+func waku_relay_enough_peers(topic *C.char, cb C.WakuCallBack, userData unsafe.Pointer) C.int {
 	return singleFnExec(func() (string, error) {
 		result, err := library.RelayEnoughPeers(C.GoString(topic))
 		if result {
 			return "true", err
 		}
 		return "false", err
-	}, onOkCb, onErrCb)
+	}, cb, userData)
 }
 
-// Publish a message using waku relay and returns the message ID. Use NULL for topic to use the default pubsub topic
+// Publish a message using waku relay and returns the message ID. Use NULL for topic to derive the pubsub topic from the contentTopic.
 // If ms is greater than 0, the broadcast of the message must happen before the timeout
 // (in milliseconds) is reached, or an error will be returned.
 //
 //export waku_relay_publish
-func waku_relay_publish(messageJSON *C.char, topic *C.char, ms C.int, onOkCb C.WakuCallBack, onErrCb C.WakuCallBack) C.int {
+func waku_relay_publish(messageJSON *C.char, topic *C.char, ms C.int, cb C.WakuCallBack, userData unsafe.Pointer) C.int {
 	return singleFnExec(func() (string, error) {
 		return library.RelayPublish(C.GoString(messageJSON), C.GoString(topic), int(ms))
```

Comment:
would the call to C.GoString be fine and not cause issue if topic is NULL?

Reply:
It doesn't cause a problem. `C.GoString(nil) == ""`